python/lsst/pipe/tasks/hips.py

```diff
@@ -604,3 +614,907 @@ def build_quantum_graph(
+        multiple=True,
+        deferLoad=True,
+    )
+
+
+class GenerateHipsConfig(pipeBase.PipelineTaskConfig,
+                         pipelineConnections=GenerateHipsConnections):
+    """Configuration parameters for GenerateHipsTask."""
+    # WARNING: In general PipelineTasks are not allowed to do any outputs
+    # outside of the butler.  This task has been given (temporary)
+    # Special Dispensation because of the nature of HiPS outputs until
+    # a more controlled solution can be found.
+    hips_base_uri = pexConfig.Field(
+        doc="URI to HiPS base for output.",
+        dtype=str,
+        optional=False,
+    )
+    min_order = pexConfig.Field(
+        doc="Minimum healpix order for HiPS tree.",
+        dtype=int,
+        default=3,
+    )
+    properties = pexConfig.ConfigField(
+        dtype=HipsPropertyConfig,
+        doc="Configuration for property file.",
+    )
+    png_gray_asinh_minimum = pexConfig.Field(
+        doc="AsinhMapping intensity to be mapped to black for grayscale png scaling.",
+        dtype=float,
+        default=0.0,
+    )
+    png_gray_asinh_stretch = pexConfig.Field(
+        doc="AsinhMapping linear stretch for grayscale png scaling.",
+        dtype=float,
+        default=2.0,
+    )
+    png_gray_asinh_softening = pexConfig.Field(
+        doc="AsinhMapping softening parameter (Q) for grayscale png scaling.",
+        dtype=float,
+        default=8.0,
+    )
+
+
+class GenerateHipsTask(pipeBase.PipelineTask):
+    """Task for making a HiPS tree with fits and grayscale pngs."""
+    ConfigClass = GenerateHipsConfig
+    _DefaultName = "generateHips"
+
+    @timeMethod
+    def runQuantum(self, butlerQC, inputRefs, outputRefs):
+        inputs = butlerQC.get(inputRefs)
+
+        dims = inputRefs.hips_exposure_handles[0].dataId.names
+        order = None
+        for dim in dims:
+            if "healpix" in dim:
+                order = int(dim.split("healpix")[1])
+                healpix_dim = dim
+        if order is None:
+            raise RuntimeError("Could not determine healpix order for input exposures.")
+
+        hips_exposure_handle_dict = {
+            (hips_exposure_handle.dataId[healpix_dim],
+             hips_exposure_handle.dataId['band']): hips_exposure_handle
+            for hips_exposure_handle in inputs["hips_exposure_handles"]
+        }
+
+        band = butlerQC.quantum.dataId['band']
+
+        self.run(bands=[band], max_order=order, hips_exposure_handle_dict=hips_exposure_handle_dict)
+
+    def run(self, bands, max_order, hips_exposure_handle_dict, do_color=False):
+        """Run the GenerateHipsTask.
+
+        Parameters
+        ----------
+        bands : `list [ `str` ]
+            List of bands to be processed (or single band).
+        max_order : `int`
+            HEALPix order of the maximum (native) HPX exposures.
+        hips_exposure_handle_dict : `dict` {`int`: `lsst.daf.butler.DeferredDatasetHandle`}
```

Comment:
This type annotation is not right but you likely don't have space to make it correct.